src/types/iterator.rs

```diff
@@ -213,4 +229,30 @@ mod tests {
         let iter: &PyIterator = PyIterator::try_from(obj.as_ref(py)).unwrap();
         assert_eq!(obj, iter.into());
     }
+
+    #[test]
+    fn test_as_ref() {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let iter: Py<PyIterator> = PyList::empty(py).to_object(py).as_ref(py).iter().unwrap().into();
+        let mut iter_ref: &PyIterator = iter.as_ref(py);
+        assert!(matches!(iter_ref.next(), Option::<PyResult::<&PyAny>>::None));
+    }
+
+    #[test]
+    fn test_into_ref() {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let l = PyList::empty(py).to_object(py);
+        let bare_ref = l.as_ref(py).iter().unwrap();
+        assert_eq!(bare_ref.get_refcnt(), 1);
+        let iter: Py<PyIterator> = bare_ref.into();
+        assert_eq!(bare_ref.get_refcnt(), 2);
+        Python::with_gil(|py| {
+            let mut iter_ref = iter.into_ref(py);
+            assert!(matches!(iter_ref.next(), Option::<PyResult::<&PyAny>>::None));
+            assert_eq!(iter_ref.get_refcnt(), 2);
+        });
```

Comment:
You already have the GIL, so no need to re-acquire

```suggestion
        let mut iter_ref = iter.into_ref(py);
        assert!(matches!(iter_ref.next(), Option::<PyResult::<&PyAny>>::None));
        assert_eq!(iter_ref.get_refcnt(), 2);
```